fix(docker-casa): couldn't retrieve remote JWKS (exceeded configured input limit of 51200 bytes)

The `jans-auth` JWKS size could grow exceeding 51200 bytes, but default size limit set by nimbusds is 51200 bytes, hence the following error will be thrown:

```
casa - 28-04 19:03:22.306 ERROR gluu.casa.core.OIDCFlowService OIDCFlowService.java:241- Couldn't retrieve remote JWK set: Exceeded configured input l
imit of 51200 bytes                                                                                                                                   
com.nimbusds.jose.RemoteKeySourceException: Couldn't retrieve remote JWK set: Exceeded configured input limit of 51200 bytes                          
        at com.nimbusds.jose.jwk.source.RemoteJWKSet.updateJWKSetFromURL(RemoteJWKSet.java:309) ~[nimbus-jose-jwt-9.28.jar:9.28]                      
        at com.nimbusds.jose.jwk.source.RemoteJWKSet.get(RemoteJWKSet.java:439) ~[nimbus-jose-jwt-9.28.jar:9.28]                                      
        at com.nimbusds.jose.proc.JWSVerificationKeySelector.selectJWSKeys(JWSVerificationKeySelector.java:157) ~[nimbus-jose-jwt-9.28.jar:9.28]      
        at com.nimbusds.jwt.proc.DefaultJWTProcessor.selectKeys(DefaultJWTProcessor.java:283) ~[nimbus-jose-jwt-9.28.jar:9.28]                        
        at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:354) ~[nimbus-jose-jwt-9.28.jar:9.28]                           
        at com.nimbusds.openid.connect.sdk.validators.IDTokenValidator.validate(IDTokenValidator.java:321) ~[oauth2-oidc-sdk-10.4.jar:10.4]           
        at com.nimbusds.openid.connect.sdk.validators.IDTokenValidator.validate(IDTokenValidator.java:254) ~[oauth2-oidc-sdk-10.4.jar:10.4]           
        at org.gluu.casa.core.OIDCFlowService.validateIDToken(OIDCFlowService.java:237) ~[classes/:?]                                                 
        at org.gluu.casa.core.OIDCFlowService.getTokens(OIDCFlowService.java:161) ~[classes/:?]                                                       
        at org.gluu.casa.core.OIDCFlowService$Proxy$_$$_WeldClientProxy.getTokens(Unknown Source) ~[classes/:?]                                       
        at org.gluu.casa.core.navigation.HomeInitiator.doInit(HomeInitiator.java:60) ~[classes/:?]                                                    
        at org.zkoss.zk.ui.metainfo.InitiatorInfo.doInit(InitiatorInfo.java:148) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]                     
        at org.zkoss.zk.ui.metainfo.InitiatorInfo.newInitiator(InitiatorInfo.java:143) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]               
        at org.zkoss.zk.ui.metainfo.PageDefinition.doInit(PageDefinition.java:398) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]                   
        at org.zkoss.zk.ui.impl.Initiators.doInit(Initiators.java:55) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.impl.UiEngineImpl.execNewPage0(UiEngineImpl.java:435) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]                     
        at org.zkoss.zk.ui.impl.UiEngineImpl.execNewPage(UiEngineImpl.java:371) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.http.DHtmlLayoutServlet.process(DHtmlLayoutServlet.java:220) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.http.DHtmlLayoutServlet.doGet(DHtmlLayoutServlet.java:146) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1419) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:527) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131) ~[?:?]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:596) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1380) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1302) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129) ~[?:?]
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:215) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:135) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.server.ResourceService.sendWelcome(ResourceService.java:449) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.server.ResourceService.doGet(ResourceService.java:256) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.servlet.DefaultServlet.doGet(DefaultServlet.java:453) ~[jetty-servlet-11.0.13.jar:11.0.13]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1410) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665) ~[?:?]
        at org.jboss.resteasy.plugins.server.servlet.FilterDispatcher.doFilter(FilterDispatcher.java:71) ~[resteasy-core-6.0.3.Final.jar:6.0.3.Final]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:202) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1635) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:527) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131) ~[?:?]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1380) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1302) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:192) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:51) ~[?:?]
        at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:173) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[?:?]
        at org.eclipse.jetty.server.Server.handle(Server.java:563) ~[?:?]
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:505) ~[?:?]
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:762) ~[?:?]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:497) ~[?:?]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282) ~[?:?]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314) ~[?:?]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100) ~[?:?]
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:416) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:385) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:272) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.lambda$new$0(AdaptiveExecutionStrategy.java:140) ~[?:?]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411) ~[?:?]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:934) ~[?:?]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1078) ~[?:?]
        at java.lang.Thread.run(Unknown Source) ~[?:?]
Caused by: java.io.IOException: Exceeded configured input limit of 51200 bytes
        at com.nimbusds.jose.util.BoundedInputStream.read(BoundedInputStream.java:131) ~[nimbus-jose-jwt-9.28.jar:9.28]
        at sun.nio.cs.StreamDecoder.readBytes(Unknown Source) ~[?:?]
        at sun.nio.cs.StreamDecoder.implRead(Unknown Source) ~[?:?]
        at sun.nio.cs.StreamDecoder.read(Unknown Source) ~[?:?]
        at java.io.InputStreamReader.read(Unknown Source) ~[?:?]
        at com.nimbusds.jose.util.IOUtils.readInputStreamToString(IOUtils.java:70) ~[nimbus-jose-jwt-9.28.jar:9.28]
        at com.nimbusds.jose.util.DefaultResourceRetriever.retrieveResource(DefaultResourceRetriever.java:265) ~[nimbus-jose-jwt-9.28.jar:9.28]
        at com.nimbusds.jose.jwk.source.RemoteJWKSet.updateJWKSetFromURL(RemoteJWKSet.java:307) ~[nimbus-jose-jwt-9.28.jar:9.28]
        ... 84 more
casa - 28-04 19:03:22.327 ERROR casa.core.navigation.HomeInitiator HomeInitiator.java:85- Couldn't retrieve remote JWK set: Exceeded configured input 
limit of 51200 bytes
        com.nimbusds.oauth2.sdk.GeneralException: Couldn't retrieve remote JWK set: Exceeded configured input limit of 51200 bytes
        at org.gluu.casa.core.OIDCFlowService.getTokens(OIDCFlowService.java:178) ~[classes/:?]
        at org.gluu.casa.core.OIDCFlowService$Proxy$_$$_WeldClientProxy.getTokens(Unknown Source) ~[classes/:?]
        at org.gluu.casa.core.navigation.HomeInitiator.doInit(HomeInitiator.java:60) ~[classes/:?]
        at org.zkoss.zk.ui.metainfo.InitiatorInfo.doInit(InitiatorInfo.java:148) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.metainfo.InitiatorInfo.newInitiator(InitiatorInfo.java:143) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.metainfo.PageDefinition.doInit(PageDefinition.java:398) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.impl.Initiators.doInit(Initiators.java:55) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.impl.UiEngineImpl.execNewPage0(UiEngineImpl.java:435) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.impl.UiEngineImpl.execNewPage(UiEngineImpl.java:371) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.http.DHtmlLayoutServlet.process(DHtmlLayoutServlet.java:220) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at org.zkoss.zk.ui.http.DHtmlLayoutServlet.doGet(DHtmlLayoutServlet.java:146) ~[zk-9.6.0.2-jakarta.jar:9.6.0.2-migrated-1.0.0]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1419) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:527) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131) ~[?:?]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:596) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1380) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1380) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1302) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129) ~[?:?]
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:215) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:135) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.server.ResourceService.sendWelcome(ResourceService.java:449) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.server.ResourceService.doGet(ResourceService.java:256) ~[jetty-server-11.0.13.jar:11.0.13]
        at org.eclipse.jetty.servlet.DefaultServlet.doGet(DefaultServlet.java:453) ~[jetty-servlet-11.0.13.jar:11.0.13]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:500) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:587) ~[jetty-jakarta-servlet-api-5.0.2.jar:?]
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1410) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665) ~[?:?]
        at org.jboss.resteasy.plugins.server.servlet.FilterDispatcher.doFilter(FilterDispatcher.java:71) ~[resteasy-core-6.0.3.Final.jar:6.0.3.Final]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:202) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1635) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:527) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131) ~[?:?]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:131) ~[?:?]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:578) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:223) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1570) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:221) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1380) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:176) ~[?:?]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:484) ~[?:?]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1543) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:174) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1302) ~[?:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:129) ~[?:?]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:192) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:51) ~[?:?]
        at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:173) ~[?:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122) ~[?:?]
        at org.eclipse.jetty.server.Server.handle(Server.java:563) ~[?:?]
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:505) ~[?:?]
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:762) ~[?:?]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:497) ~[?:?]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282) ~[?:?]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314) ~[?:?]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100) ~[?:?]
        at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:416) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:385) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:272) ~[?:?]
        at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.lambda$new$0(AdaptiveExecutionStrategy.java:140) ~[?:?]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411) ~[?:?]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:934) ~[?:?]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1078) ~[?:?]
        at java.lang.Thread.run(Unknown Source) ~[?:?]
```

--- a/docker-casa/Dockerfile
+++ b/docker-casa/Dockerfile
@@ -34,8 +34,8 @@ EXPOSE 8080
 # Casa
 # ====
 
-ENV GLUU_VERSION=5.0.0-12
-ENV GLUU_BUILD_DATE='2023-04-18 13:02'
+ENV GLUU_VERSION=5.0.0-SNAPSHOT
+ENV GLUU_BUILD_DATE='2023-04-10 15:24'
 ENV GLUU_SOURCE_URL=https://jenkins.gluu.org/maven/org/gluu/casa/${GLUU_VERSION}/casa-${GLUU_VERSION}.war
 
 # Install Casa
@@ -96,7 +96,7 @@ RUN cd /tmp/jans \
 # flex setup sync
 # ===============
 
-ENV FLEX_VERSION=257d2c9c4c8f4dad6bce20634efb090678a3c6bb
+ENV FLEX_VERSION=96d82038fe752cfff4e722113c757978b2b39119
 ARG FLEX_SETUP_DIR=flex-linux-setup/flex_linux_setup
 ARG CASA_EXTRAS_DIR=casa/extras
 
@@ -109,7 +109,8 @@ RUN git clone --filter blob:none --no-checkout https://github.com/GluuFederation
     && git sparse-checkout add ${FLEX_SETUP_DIR} \
     && cp ${FLEX_SETUP_DIR}/templates/casa_*.ldif /app/templates \
     && git sparse-checkout add ${CASA_EXTRAS_DIR} \
-    && cp ${CASA_EXTRAS_DIR}/Casa.py /app/static/extension/person_authentication/
+    && cp ${CASA_EXTRAS_DIR}/Casa.py /app/static/extension/person_authentication/ \
+    && cp ${FLEX_SETUP_DIR}/templates/casa_scopes.json /app/static
 
 # =======
 # Cleanup
@@ -253,7 +254,8 @@ RUN chmod -R g=u ${JETTY_BASE}/casa/static \
     && chmod 664 /opt/jetty/etc/webdefault.xml \
     && chown -R 1000:0 ${JETTY_BASE}/common/libs \
     && chown -R 1000:0 /usr/share/java \
-    && chown -R 1000:0 /opt/prometheus
+    && chown -R 1000:0 /opt/prometheus \
+    && chmod -R g=u /app/templates
 
 USER 1000
 

--- a/docker-casa/Dockerfile
+++ b/docker-casa/Dockerfile
@@ -207,7 +207,8 @@ ENV CN_MAX_RAM_PERCENTAGE=75.0 \
     CN_GOOGLE_SECRET_NAME_PREFIX=jans \
     GLUU_CASA_ADMIN_LOCK_FILE=/opt/jans/jetty/casa/resources/.administrable \
     CN_PROMETHEUS_PORT="" \
-    CN_VERSION=1.0.13-SNAPSHOT
+    CN_VERSION=1.0.13-SNAPSHOT \
+    GLUU_CASA_JWKS_SIZE_LIMIT=100000
 
 # ==========
 # misc stuff

--- a/docker-casa/README.md
+++ b/docker-casa/README.md
@@ -76,6 +76,7 @@ The following environment variables are supported by the container:
 - `GLUU_CASA_APP_LOGGERS`: Custom logging configuration in JSON-string format with hash type (see [Configure app loggers](#configure-app-loggers) section for details).
 - `GLUU_CASA_ADMIN_LOCK_FILE`: Path to lock file to enable/disable administration feature (default to `/opt/jans/jetty/casa/resources/.administrable`). If file is not exist, the feature is disabled.
 - `CN_PROMETHEUS_PORT`: Port used by Prometheus JMX agent (default to empty string). To enable Prometheus JMX agent, set the value to a number. See [Exposing metrics](#exposing-metrics) for details.
+- `GLUU_CASA_JWKS_SIZE_LIMIT`: Default HTTP size limit (in bytes) when retrieving remote JWKS (default to `100000`).
 
 ### Configure app loggers
 

--- a/docker-casa/scripts/bootstrap.py
+++ b/docker-casa/scripts/bootstrap.py
@@ -7,6 +7,8 @@ from uuid import uuid4
 from string import Template
 from functools import cached_property
 
+from ldif import LDIFWriter
+
 from jans.pycloudlib import get_manager
 from jans.pycloudlib.persistence import render_couchbase_properties
 from jans.pycloudlib.persistence import render_base_properties
@@ -264,20 +266,23 @@ class PersistenceSetup:
         # java.lang.IllegalStateException: Duplicate key casa (attempted merging values 1 and 1)
         if not self._deprecated_script_exists():
             filenames.append("casa_person_authentication_script.ldif")
+
+        # generate extra scopes
+        self.generate_scopes_ldif()
+        filenames.append("casa_scopes.ldif")
+
         return [f"/app/templates/{filename}" for filename in filenames]
 
     def _deprecated_script_exists(self):
-        script_exists = False
-
         # deprecated Casa script DN
         id_ = "inum=BABA-CACA,ou=scripts,o=jans"
 
         # sql and spanner
         if self.persistence_type in ("sql", "spanner"):
-            script_exists = bool(self.client.get("jansCustomScr", doc_id_from_dn(id_)))
+            return bool(self.client.get("jansCustomScr", doc_id_from_dn(id_)))
 
         # couchbase
-        elif self.persistence_type == "couchbase":
+        if self.persistence_type == "couchbase":
             bucket = os.environ.get("CN_COUCHBASE_BUCKET_PREFIX", "jans")
             key = id_from_dn(id_)
             req = self.client.exec_query(
@@ -285,18 +290,42 @@ class PersistenceSetup:
             )
             with contextlib.suppress(IndexError):
                 entry = req.json()["results"][0]
-                script_exists = bool(entry["id"])
+                return bool(entry["id"])
 
         # ldap
-        else:
-            script_exists = bool(self.client.get(id_))
-
-        return script_exists
+        return bool(self.client.get(id_))
 
     def import_ldif_files(self):
         for file_ in self.ldif_files:
             logger.info(f"Importing {file_}")
             self.client.create_from_ldif(file_, self.ctx)
+
+    def generate_scopes_ldif(self):
+        # prepare required scopes (if any)
+        with open("/app/static/casa_scopes.json") as f:
+            scopes = json.loads(f.read())
+
+        with open("/app/templates/casa_scopes.ldif", "wb") as fd:
+            writer = LDIFWriter(fd, cols=1000)
+
+            for scope in scopes:
+                writer.unparse(
+                    f"inum={scope['inum']},ou=scopes,o=jans",
+                    {
+                        "objectClass": ["top", "jansScope"],
+                        "description": [scope["description"]],
+                        "displayName": [scope["displayName"]],
+                        "inum": [scope["inum"]],
+                        "jansDefScope": [str(scope["jansDefScope"]).lower()],
+                        "jansId": [scope["jansId"]],
+                        "jansScopeTyp": [scope["jansScopeTyp"]],
+                        "jansAttrs": [json.dumps({
+                            "spontaneousClientId": None,
+                            "spontaneousClientScopes": [],
+                            "showInConfigurationEndpoint": False,
+                        })],
+                    },
+                )
 
 
 if __name__ == "__main__":

--- a/docker-casa/scripts/entrypoint.sh
+++ b/docker-casa/scripts/entrypoint.sh
@@ -69,6 +69,7 @@ exec java \
     -Djava.io.tmpdir=/tmp \
     -Dlog4j2.configurationFile=resources/log4j2.xml \
     -Dadmin.lock=${GLUU_CASA_ADMIN_LOCK_FILE} \
+    -Dcom.nimbusds.jose.jwk.source.RemoteJWKSet.defaultHttpSizeLimit=${GLUU_CASA_JWKS_SIZE_LIMIT} \
     $(get_prometheus_opt) \
     ${CN_JAVA_OPTIONS} \
     -jar /opt/jetty/start.jar jetty.httpConfig.sendServerVersion=false jetty.deploy.scanInterval=0

--- a/docker-casa/scripts/entrypoint.sh
+++ b/docker-casa/scripts/entrypoint.sh
@@ -48,6 +48,7 @@ get_prometheus_lib() {
 get_prometheus_lib
 python3 /app/scripts/wait.py
 python3 /app/scripts/bootstrap.py
+python3 /app/scripts/upgrade.py
 # python3 /app/scripts/jca_sync.py &
 python3 /app/scripts/auth_conf.py
 python3 /app/scripts/mod_context.py casa

--- a/docker-casa/scripts/upgrade.py
+++ b/docker-casa/scripts/upgrade.py
@@ -1,0 +1,222 @@
+import json
+import logging.config
+import os
+from collections import namedtuple
+
+from jans.pycloudlib import get_manager
+from jans.pycloudlib.persistence import CouchbaseClient
+from jans.pycloudlib.persistence import LdapClient
+from jans.pycloudlib.persistence import SpannerClient
+from jans.pycloudlib.persistence import SqlClient
+from jans.pycloudlib.persistence import PersistenceMapper
+from jans.pycloudlib.persistence import doc_id_from_dn
+from jans.pycloudlib.persistence import id_from_dn
+
+from settings import LOGGING_CONFIG
+
+logging.config.dictConfig(LOGGING_CONFIG)
+logger = logging.getLogger("entrypoint")
+
+Entry = namedtuple("Entry", ["id", "attrs"])
+
+
+class LDAPBackend:
+    def __init__(self, manager):
+        self.manager = manager
+        self.client = LdapClient(manager)
+        self.type = "ldap"
+
+    def format_attrs(self, attrs):
+        _attrs = {}
+        for k, v in attrs.items():
+            if len(v) < 2:
+                v = v[0]
+            _attrs[k] = v
+        return _attrs
+
+    def get_entry(self, key, filter_="", attrs=None, **kwargs):
+        filter_ = filter_ or "(objectClass=*)"
+
+        entry = self.client.get(key, filter_=filter_, attributes=attrs)
+        if not entry:
+            return None
+        return Entry(entry.entry_dn, self.format_attrs(entry.entry_attributes_as_dict))
+
+    def modify_entry(self, key, attrs=None, **kwargs):
+        attrs = attrs or {}
+        del_flag = kwargs.get("delete_attr", False)
+
+        if del_flag:
+            mod = self.client.MODIFY_DELETE
+        else:
+            mod = self.client.MODIFY_REPLACE
+
+        for k, v in attrs.items():
+            if not isinstance(v, list):
+                v = [v]
+            attrs[k] = [(mod, v)]
+        return self.client.modify(key, attrs)
+
+
+class SQLBackend:
+    def __init__(self, manager):
+        self.manager = manager
+        self.client = SqlClient(manager)
+        self.type = "sql"
+
+    def get_entry(self, key, filter_="", attrs=None, **kwargs):
+        table_name = kwargs.get("table_name")
+        entry = self.client.get(table_name, key, attrs)
+
+        if not entry:
+            return None
+        return Entry(key, entry)
+
+    def modify_entry(self, key, attrs=None, **kwargs):
+        attrs = attrs or {}
+        table_name = kwargs.get("table_name")
+        return self.client.update(table_name, key, attrs), ""
+
+
+class CouchbaseBackend:
+    def __init__(self, manager):
+        self.manager = manager
+        self.client = CouchbaseClient(manager)
+        self.type = "couchbase"
+
+    def get_entry(self, key, filter_="", attrs=None, **kwargs):
+        bucket = kwargs.get("bucket")
+        req = self.client.exec_query(
+            f"SELECT META().id, {bucket}.* FROM {bucket} USE KEYS '{key}'"
+        )
+        if not req.ok:
+            return None
+
+        try:
+            _attrs = req.json()["results"][0]
+            id_ = _attrs.pop("id")
+            entry = Entry(id_, _attrs)
+        except IndexError:
+            entry = None
+        return entry
+
+    def modify_entry(self, key, attrs=None, **kwargs):
+        bucket = kwargs.get("bucket")
+        del_flag = kwargs.get("delete_attr", False)
+        attrs = attrs or {}
+
+        if del_flag:
+            kv = ",".join(attrs.keys())
+            mod_kv = f"UNSET {kv}"
+        else:
+            kv = ",".join([
+                "{}={}".format(k, json.dumps(v))
+                for k, v in attrs.items()
+            ])
+            mod_kv = f"SET {kv}"
+
+        query = f"UPDATE {bucket} USE KEYS '{key}' {mod_kv}"
+        req = self.client.exec_query(query)
+
+        if req.ok:
+            resp = req.json()
+            status = bool(resp["status"] == "success")
+            message = resp["status"]
+        else:
+            status = False
+            message = req.text or req.reason
+        return status, message
+
+
+class SpannerBackend:
+    def __init__(self, manager):
+        self.manager = manager
+        self.client = SpannerClient(manager)
+        self.type = "spanner"
+
+    def get_entry(self, key, filter_="", attrs=None, **kwargs):
+        table_name = kwargs.get("table_name")
+        entry = self.client.get(table_name, key, attrs)
+
+        if not entry:
+            return None
+        return Entry(key, entry)
+
+    def modify_entry(self, key, attrs=None, **kwargs):
+        attrs = attrs or {}
+        table_name = kwargs.get("table_name")
+        return self.client.update(table_name, key, attrs), ""
+
+
+BACKEND_CLASSES = {
+    "sql": SQLBackend,
+    "couchbase": CouchbaseBackend,
+    "spanner": SpannerBackend,
+    "ldap": LDAPBackend,
+}
+
+
+class Upgrade:
+    def __init__(self, manager):
+        self.manager = manager
+
+        mapper = PersistenceMapper()
+
+        backend_cls = BACKEND_CLASSES[mapper.mapping["default"]]
+        self.backend = backend_cls(manager)
+
+    def invoke(self):
+        logger.info("Running upgrade process (if required)")
+        self.update_client_scopes()
+
+    def update_client_scopes(self):
+        kwargs = {}
+        client_id = self.manager.config.get("casa_client_id")
+        id_ = f"inum={client_id},ou=clients,o=jans"
+
+        if self.backend.type in ("sql", "spanner"):
+            kwargs = {"table_name": "jansClnt"}
+            id_ = doc_id_from_dn(id_)
+        elif self.backend.type == "couchbase":
+            kwargs = {"bucket": os.environ.get("CN_COUCHBASE_BUCKET_PREFIX", "jans")}
+            id_ = id_from_dn(id_)
+
+        entry = self.backend.get_entry(id_, **kwargs)
+
+        if not entry:
+            return
+
+        if self.backend.type == "sql" and self.backend.client.dialect == "mysql":
+            client_scopes = entry.attrs["jansScope"]["v"]
+        else:
+            client_scopes = entry.attrs["jansScope"]
+
+        if not isinstance(client_scopes, list):
+            client_scopes = [client_scopes]
+
+        # all potential new scopes for client
+        with open("/app/static/casa_scopes.json") as f:
+            new_client_scopes = [
+                f"inum={scope['inum']},ou=scopes,o=jans"
+                for scope in json.loads(f.read())
+            ]
+
+        # find missing scopes from the client
+        diff = list(set(new_client_scopes).difference(client_scopes))
+
+        if diff:
+            if self.backend.type == "sql" and self.backend.client.dialect == "mysql":
+                entry.attrs["jansScope"]["v"] = client_scopes + diff
+            else:
+                entry.attrs["jansScope"] = client_scopes + diff
+            self.backend.modify_entry(entry.id, entry.attrs, **kwargs)
+
+
+def main():
+    manager = get_manager()
+    upgrade = Upgrade(manager)
+    upgrade.invoke()
+
+
+if __name__ == "__main__":
+    main()

